### DISCUSSION TITLE
[0.23.1] Custom record Query

### DISCUSSION THREAD

**Discussion by antonkad** [pocketbase/pocketbase#5958] [Q&A]
Hello, 

Since the update record query does not populate my custom interface. I returns an empty array.

```go
	projects := []*kadmodel.Project{}
	err := app.RecordQuery("projects").
		All(&records)

...

package kadmodel

type Project struct {
	ID              string          `db:"id" json:"id"`
	Name            string          `db:"name" json:"name"`
}
```

The only information I go from the doc is that i should use ```records := []*core.Record{}``` instead, it does put the record results in this variable.
but I don't know how I can do as I did previously in order to access my variables such as ```projects.id```

Anton

**Comment by ganigeorgiev** [comment]
I'll consider adding a fallback for custom structs with the next minor release but for now you can replace `.RecordQuery().` with `.DB().Select("*").From("projects").`.

An alternative to your custom struct could be to convert it to a `RecordProxy` - https://pocketbase.io/docs/go-record-proxy/.

**Reply by ganigeorgiev** [reply]
_I've pushed a fix in the `develop` branch to allow `RecordQuery().All()` scanning into custom structs._

**Reply by antonkad** [reply]
Thanks for the solution: 
``` .DB().Select("*").From("projects").``` quickfix works
I will explore the recordProxy later on.

**Reply by ganigeorgiev** [reply]
As mentioned above, the custom struct scanning should also work with the `develop` branch (_I'll push a release sometime later this week_), aka. you can update your `go.mod` manually or try `go get github.com/pocketbase/pocketbase@develop` if you don't want to change the `RecordQuery()` calls.

**Comment by antonkad** [comment]
I started implementing RecordProxy, I was wondering how to work with relation fields and how to expand a proxy record ?

```
func (p *Project) FrameworkPreset(app *pocketbase.PocketBase) *FrameworkPreset {
	preset := &FrameworkPreset{}
	errs := app.ExpandRecord(p.Record, []string{"frameworkPreset"}, nil)
	if len(errs) > 0 {
		fmt.Errorf("failed to expand: %v", errs)
	}
	return preset
}
func (p *Project) SetFrameworkPreset(frameworkPreset *FrameworkPreset) {
	p.Set("frameworkPreset", frameworkPreset)
}

```

**Reply by ganigeorgiev** [reply]
Don't overcomplicate your setup. There is no mechanism at the moment that will load/store the expanded relation records as the `RecordProxy` out of the box. Note that expanding relations will load/store the relations into the specified record and could have unintended side-effects when serializing or even cause out-of-sync issues in case the "frameworkPreset" field changes after the expand which may not be what you are trying to do.

Additionally, calling `p.Set("frameworkPreset", frameworkPreset)` is wrong because it is expected the setter to operate with the related record id and not with the entire model (aka. you need to update the setter to use `frameworkPreset.Id`).

For the getter, you can construct an individual record query that will return your relation record, aka. something like:

```go
func (p *Project) FrameworkPreset(app core.App) (*FrameworkPreset, error) {
    preset := &FrameworkPreset{}

    err := app.RecordQuery("frameworkPresets").
        AndWhere(dbx.HashExp{
            "id": p.GetString("frameworkPreset"),
        }).
        Limit(1).
        One(preset)

    if err != nil {
        return nil, err
    }

    return preset, nil
}
```

**Reply by antonkad** [reply]
Understood ! I was exploring the various available functions.

Thanks for the quick reply and detailed answer
